test/dependency_source/test_requirement.py

```diff
@@ -378,109 +427,30 @@ def test_requirement_source_require_hashes_unpinned(req_file):
         list(source.collect())
 
 
-def test_requirement_source_no_deps(req_file):
-    source = _init_requirement([(req_file(), "flask==2.0.1")], no_deps=True)
-
-    specs = list(source.collect())
-    assert specs == [ResolvedDependency("flask", Version("2.0.1"))]
-
-
-def test_requirement_source_no_deps_unpinned(req_file):
-    source = _init_requirement([(req_file(), "flask\nrequests==1.0")], no_deps=True)
-
-    # When dependency resolution is disabled, all requirements must be pinned.
-    with pytest.raises(DependencySourceError):
-        list(source.collect())
-
-
-def test_requirement_source_no_deps_not_exact_version(req_file):
-    source = _init_requirement([(req_file(), "flask==1.0\nrequests>=1.0")], no_deps=True)
-
-    # When dependency resolution is disabled, all requirements must be pinned.
-    with pytest.raises(DependencySourceError):
-        list(source.collect())
-
-
-def test_requirement_source_no_deps_unpinned_url(req_file):
-    source = _init_requirement(
-        [
-            (
-                req_file(),
-                "https://github.com/pallets/flask/archive/refs/tags/2.0.1.tar.gz#egg=flask\n",
-            )
-        ],
-        no_deps=True,
-    )
-
-    assert list(source.collect()) == [
-        SkippedDependency(
-            name="flask",
-            skip_reason="URL requirements cannot be pinned to a specific package version",
-        )
-    ]
-
-
-def test_requirement_source_no_deps_editable_with_egg_fragment(req_file):
-    source = _init_requirement([(req_file(), "-e file:flask.py#egg=flask==2.0.1")], no_deps=True)
-
-    specs = list(source.collect())
-    assert (
-        SkippedDependency(
-            name="flask",
-            skip_reason="URL requirements cannot be pinned to a specific package version",
-        )
-        in specs
-    )
-
-
-def test_requirement_source_no_deps_editable_without_egg_fragment(req_file):
-    source = _init_requirement([(req_file(), "-e file:flask.py")], no_deps=True)
-
-    specs = list(source.collect())
-    assert (
-        SkippedDependency(
-            name="-e file:flask.py",
-            skip_reason="could not deduce package version from URL requirement",
-        )
-        in specs
-    )
-
-
-def test_requirement_source_no_deps_non_editable_without_egg_fragment(req_file):
+def test_requirement_source_require_hashes_incorrect_hash(req_file):
```

Comment:
I added a few tests but to be honest, some of these (and the existing ones) are essentially testing `pip`. We could potentially trim these down to a bare minimum that at least confirms that we're providing the right flags to `pip` and call it a day.

Reply:
Yeah, I'd be okay with dropping tests that aren't contributing to our coverage. But yes, it'd be good to confirm that we pass the right flags in some new tests.